packages/fxa-content-server/server/lib/routes/validate-email-domain.js

```diff
@@ -0,0 +1,62 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * This is basically a RPC endpoint for the frontend to resolve a domain to see
+ * if it has MX record, A record, or neither.
+ */
+
+'use strict';
+
+const joi = require('joi');
+const dns = require('dns');
+const resolver = new dns.promises.Resolver();
+const results = ['MX', 'A', 'none'].reduce((accumulator, val) => {
+  accumulator[val] = val;
+  return accumulator;
+}, {});
+const NotFoundErrorCodes = [dns.NODATA, dns.NOTFOUND];
+
+const tryResolveWith = resolveFunc => async domain => {
+  try {
+    const records = await resolveFunc(domain);
+    // We don't do anything with the records
+    return records && records.length;
+  } catch (err) {
+    if (NotFoundErrorCodes.includes(err.code)) {
+      return false;
+    }
+    throw err;
+  }
+};
+
+module.exports = function() {
+  return {
+    method: 'post',
```

Comment:
Semantically, a POST seems strange here since we aren't actually creating/updating any records. How come POST instead of a GET? Is it to keep the domain name from being logged? If so, can you leave a comment about why?

Reply:
I viewed it as a RPC instead of a RESTful API, so I did not place a lot of emphasis on the HTTP method.  Sending payloads in the body feels "cleaner".  Obviously it was the wrong choice; if it confused you then it will confuse others.  I will change it to GET. 👍